podcasts/SyncHashTable.swift

```diff
@@ -0,0 +1,45 @@
+class SyncHashTable<Key: Hashable, Value> {
+
+    private let queue: DispatchQueue
+    private var table: [Key: Value]
+
+    init(label: String = UUID().uuidString) {
+        queue = DispatchQueue(label: label)
+        table = [:]
+    }
+
+    func value(forKey key: Key) -> Value? {
+        var value: Value?
+        queue.sync {
+            value = table[key]
+        }
+        return value
+    }
+
+    func updateValue(_ value: Value?, forKey key: Key) {
+        queue.sync {
```

Comment:
This should be async with a barrier flag (ref. [here](https://amrit-pandey.medium.com/threadsafe-swift-collections-c90a94c0d269))

Reply:
Done [here](https://github.com/Automattic/pocket-casts-ios/pull/2748/commits/9b4ad76b3099e893f2063f1b1beea5d38ada6a1a)